### PR TITLE
git-icdiff: fix issues reported by shellcheck

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

File: git-icdiff [hunk 1]
```diff
@@ -10,8 +10,8 @@ if [ -z "$GITPAGER" ]; then
   GITPAGER="less"
 fi
 
-if [ "$GITPAGER" = "more" -o "$GITPAGER" = "less" ]; then
+if [ "$GITPAGER" = "more" ] -o [ "$GITPAGER" = "less" ]; then
```
Comment (jeffkaufman) [inline git-icdiff]: This doesn't make sense.  `-o` only works inside `[...]`.

I guess we could switch it to `if [ ... ] || [ ... ]` but the current version is fine.

File: git-icdiff [hunk 2]
```diff
@@ -10,8 +10,8 @@ if [ -z "$GITPAGER" ]; then
...
   GITPAGER="$GITPAGER -R"
 fi
 
-git difftool --no-prompt --extcmd="icdiff $ICDIFF_OPTIONS $@" | $GITPAGER
+git difftool --no-prompt --extcmd="icdiff $ICDIFF_OPTIONS $*" | $GITPAGER
```
Comment (jeffkaufman) [inline git-icdiff]: What's the advantage of using `$*` here?  I can't think of anything this fixes.

File: git-icdiff (hunk 1 above)
Comment (blueyed) [reply git-icdiff]: Oops, amended.

**Comment by blueyed** [comment]
The output from shellcheck:
```

In git-icdiff line 13:
if [ "$GITPAGER" = "more" -o "$GITPAGER" = "less" ]; then
                          ^-- SC2166: Prefer [ p ] || [ q ] as [ p -o q ] is not well defined.


In git-icdiff line 17:
git difftool --no-prompt --extcmd="icdiff $ICDIFF_OPTIONS $@" | $GITPAGER
                                                          ^-- SC2145: Argument mixes string and array. Use * or separate argument.
```

**Pull request merged**